### PR TITLE
Transition to a file-base configuration setup

### PR DESCRIPTION
**Pull request by austenmc**
This is just easier than an environment-based one. We'll see how it goes...

### CODE REVIEW

**Comment by Vassyli** [comment]
lgtm.
We maybe want to move away from ENV completely and make the configuration file path defineable (at least as an alternative) in the bootstrap directly. Currently this is what I'm doing manually in crate-www since env variables will not load (at least not on my system). This would give us the ability to store daenerys config together with symfony config (if we'd like to?).

File: src/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        $realLogPath = dirname($configFilePath) . DIRECTORY_SEPARATOR . $logPath;
+        if ($realLogPath === false || strlen($realLogPath) == 0 || is_dir($realLogPath) === false) {
+            throw new InvalidConfigurationException("Invalid or missing log path: '{$realLogPath}'.");
+        }
+        $this->logPath = $realLogPath;
+
+        $dsn = $rawConfig['database']['dsn'] ?? '';
+        $user = $rawConfig['database']['user'] ?? '';
+        $passwd = $rawConfig['database']['password'] ?? '';
+        $name = $rawConfig['database']['name'] ?? '';
```
Comment (Vassyli) [inline src/Configuration.php:48]: <3 PHP7

**Comment by austenmc** [comment]
I was thinking of defining a default config path to address this issue, but yeah let's keep thinking about it and evolving it.

What do you mean environment variables arent working?

**Comment by Vassyli** [comment]
Defining a default config path inside of the core is difficult since that would be relative to the vendor root, not the crate root.

Could be a windows issue, values from .env are not loading. That's why I'm reading the file and storing everything via putenv into the environments.

**Pull request merged**